app/src/types.ts

```diff
@@ -0,0 +1,81 @@
+import { Dispatch, SetStateAction } from 'react'
+import { UrlObject } from 'url'
+
+/*
+ * Types relating to persistent user data
+ */
+export type Clinic = {
+  id: number
+  agency: string
+  agencyAddress: string
+  agencyTelephone: string
+  clinic: string
+  clinicAddress: string
+  clinicTelephone: string
+  county: string
+  zip: string
+}
+
+export type ChooseClinicData = {
+  clinic: Clinic | undefined
+  zipCode: string
+}
+
+export type ContactData = {
+  firstName: string
+  lastName: string
+  phone: string
+  comments: string
+}
+
+export type EligibilityData = {
+  residential: string
+  categorical: string[]
+  previouslyEnrolled: string
+  adjunctive: string[]
+}
+
+export type IncomeData = {
+  householdSize: string
+}
+
+export type SessionData = {
+  chooseClinic: ChooseClinicData
+  contact: ContactData
+  eligibility: EligibilityData
+  income: IncomeData
+  submitted: boolean
+}
+
+export interface ReadSession {
+  session: SessionData
+}
+
+export interface WriteSession extends ReadSession {
+  setSession: Dispatch<SetStateAction<SessionData>> | (() => unknown)
+}
+
+export interface ClearSession extends WriteSession {
+  sessionKey: string
+}
+
+/**
+ * Types relating to page routing.
+ */
+export interface Page {
+  backRoute?: string
+  forwardRoute?: UrlObject | string
+  reviewMode?: boolean
+}
```

Comment:
Notes for reviewers: 
- Most pages in the form wizard display a "back" link, which links to the route in `backRoute`
- Most pages in the form wizard display a call to action button, which links to the route in `forwardRoute`
- Some pages in the form wizard, specifically `/eligibility`, `/income`, `/choose-clinic`, `/contact` require user input. The user is allowed to edit that input when they are in "review mode". This prop tells the next.js page whether or not it is in review mode.